wandb/__init__.py

```diff
@@ -358,9 +361,12 @@ def _init_jupyter(run):
     os.environ[env.JUPYTER] = "true"
 
     if not run.api.api_key:
-        _jupyter_login()
+        key = _jupyter_login()
         # Ensure our api client picks up the new key
-        run.api.reauth()
+        if key:
+            run.api.reauth()
+        else:
+            run.mode = "dryrun"
```

Comment:
Setting dryrun, but the code that follows assumes the run was created and suggests to the user that it is viewable in the cloud?